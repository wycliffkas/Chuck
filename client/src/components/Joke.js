import React from "react";
import Modal from "react-bootstrap/Modal";

const Joke = ({ joke, onHideModal, modal }) => {
  return (
    <>
      <Modal show={modal} onHide={onHideModal}>
        <Modal.Header closeButton>
          <Modal.Title>Joke</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <p>{joke ? joke : <p>Loading...</p>}</p>
        </Modal.Body>
        <Modal.Footer>
          <button className="btn btn-secondary" onClick={onHideModal}>
            Close
          </button>
        </Modal.Footer>
      </Modal>
    </>
  );
};

export default Joke;
