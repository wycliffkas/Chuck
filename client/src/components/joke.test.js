import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Joke from "./Joke";

configure({ adapter: new Adapter() });

describe("Joke component", () => {
  it("renders", () => {
    const wrapper = shallow(<Joke />);
    expect(wrapper).toMatchSnapshot();
  });
});