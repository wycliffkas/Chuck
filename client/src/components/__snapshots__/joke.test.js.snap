// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Joke component renders 1`] = `ShallowWrapper {}`;
